src/topics.js

```diff
@@ -327,6 +329,14 @@ var async = require('async'),
 		});
 	}
 
+	Topics.getUserBookmark = function (tid, uid, callback) {
+		Topics.getTopicField(tid + ':bookmarks', uid, callback);
+	}
+
+	Topics.setUserBookmark = function(data, callback) {
+		Topics.setTopicField(data.tid + ':bookmarks', data.uid, data.postIndex, callback);
+	}
+
```

Comment:
Do you mean instead of calling `Topics.setTopicField(tid + ':bookmarks', uid, postIndex, ...)`, make it call `Topics.setObjectField('topic:' + tid + ':bookmarks', postIndex, ... )` ?